### DISCUSSION TITLE
Relativistic Molecular Orbital Visualization

### DISCUSSION THREAD

**Discussion by DanielBajac** [pyscf/pyscf#2661] [Q&A]
Hi everyone,

I'm currently working on visualizing 4-component SCF molecular orbitals. I've attempted using both molden and cube file formats, but both seem to require real numbers. Converting complex molecular coefficients to real numbers, some orbitals become unvisualizable.

Interestingly, I found that if I only save either the small or large components in the molden file, more orbitals become accessible for visualization. And if I subdivided the small or large component, more orbitals can be visualized.

Do any of you have recommendations for plotting relativistic molecular orbitals?

Best regards,
Daniel

**Comment by xubwa** [comment]
The relativistic mo cannot be trivially visualized using a nonrelativistic visualization package. The 4c calculation in pyscf uses a j-adapted basis intrinsicly. Thus even if you found the coefficients to be real, they are real under this spinor basis, not the conventional real gto basis that visualization packages work with. Extra care need to be taken.
Instead of plotting the complex valued spinors, density might worth a look and pyscf has the necessary functions to achieve this.
As far as I know, there's no open access tool to do this. DIRAC ships with a visualization tool by plotting the density. There's also a more recent paper https://pubs.acs.org/doi/full/10.1021/acs.jpca.9b01134 by Shiozaki group and implemented in maybe their development branch of bagel.
Best,
Xubo

**Comment by DanielBajac** [comment]
Thanks a lot, Xubo. 

Best

**Comment by DanielBajac** [comment]
> Instead of plotting the complex valued spinors, density might worth a look and pyscf has the necessary functions to achieve this.

@xubwa How do you recommend to calculate electronic density of molecular orbital in relativistic regime?
